--- _csound_builtins.py	2016-06-14 19:51:50.000000000 -0700
+++ _csound_builtins.py	2016-06-14 19:48:53.000000000 -0700
@@ -1,50 +1,31 @@
 # -*- coding: utf-8 -*-
-"""
-    pygments.lexers._csound_builtins
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    :copyright: Copyright 2006-2015 by the Pygments team, see AUTHORS.
-    :license: BSD, see LICENSE for details.
-"""
-
-# Opcodes in Csound 6.05 from
+# Opcodes in Csound 6.07 (16 May 2016) from
 #   csound --list-opcodes
-# except
-#   cggoto  <http://www.csounds.com/manual/html/cggoto.html>
-#   cigoto  <http://www.csounds.com/manual/html/cigoto.html>
-#   cingoto (undocumented)
-#   ckgoto  <http://www.csounds.com/manual/html/ckgoto.html>
-#   cngoto  <http://www.csounds.com/manual/html/cngoto.html>
-#   endin   <http://www.csounds.com/manual/html/endin.html
-#   endop   <http://www.csounds.com/manual/html/endop.html
-#   goto    <http://www.csounds.com/manual/html/goto.html>
-#   igoto   <http://www.csounds.com/manual/html/igoto.html>
-#   instr   <http://www.csounds.com/manual/html/instr.html>
-#   kgoto   <http://www.csounds.com/manual/html/kgoto.html>
-#   loop_ge <http://www.csounds.com/manual/html/loop_ge.html>
-#   loop_gt <http://www.csounds.com/manual/html/loop_gt.html>
-#   loop_le <http://www.csounds.com/manual/html/loop_le.html>
-#   loop_lt <http://www.csounds.com/manual/html/loop_lt.html>
-#   opcode  <http://www.csounds.com/manual/html/opcode.html>
-#   return  <http://www.csounds.com/manual/html/return.html>
-#   rigoto  <http://www.csounds.com/manual/html/rigoto.html>
-#   tigoto  <http://www.csounds.com/manual/html/tigoto.html>
-#   timout  <http://www.csounds.com/manual/html/timout.html>
-# which are treated as keywords; the scoreline opcodes
-#   scoreline <http://www.csounds.com/manual/html/scoreline.html>
-#   scoreline_i <http://www.csounds.com/manual/html/scoreline_i.html>
-# which allow Csound Score highlighting; the pyrun opcodes
-# <http://www.csounds.com/manual/html/pyrun.html> 
-#   pylrun
-#   pylruni
-#   pylrunt
-#   pyrun
-#   pyruni
-#   pyrunt
-# which allow Python highlighting; and the Lua opcodes
-#   lua_exec  <http://www.csounds.com/manual/html/lua_exec.html>
-#   lua_opdef <http://www.csounds.com/manual/html/lua_opdef.html>
-# which allow Lua highlighting.
+# plus some deprecated opcodes, and minus
+#   cggoto   <https://csound.github.io/docs/manual/cggoto.html>
+#   cigoto   <https://csound.github.io/docs/manual/cigoto.html>
+#   cingoto  (undocumented)
+#   ckgoto   <https://csound.github.io/docs/manual/ckgoto.html>
+#   cngoto   <https://csound.github.io/docs/manual/cngoto.html>
+#   cnkgoto  (undocumented)
+#   endin    <https://csound.github.io/docs/manual/endin.html
+#   endop    <https://csound.github.io/docs/manual/endop.html
+#   goto     <https://csound.github.io/docs/manual/goto.html>
+#   igoto    <https://csound.github.io/docs/manual/igoto.html>
+#   instr    <https://csound.github.io/docs/manual/instr.html>
+#   kgoto    <https://csound.github.io/docs/manual/kgoto.html>
+#   loop_ge  <https://csound.github.io/docs/manual/loop_ge.html>
+#   loop_gt  <https://csound.github.io/docs/manual/loop_gt.html>
+#   loop_le  <https://csound.github.io/docs/manual/loop_le.html>
+#   loop_lt  <https://csound.github.io/docs/manual/loop_lt.html>
+#   opcode   <https://csound.github.io/docs/manual/opcode.html>
+#   return   <https://csound.github.io/docs/manual/return.html>
+#   rireturn <https://csound.github.io/docs/manual/rireturn.html>
+#   rigoto   <https://csound.github.io/docs/manual/rigoto.html>
+#   tigoto   <https://csound.github.io/docs/manual/tigoto.html>
+#   timout   <https://csound.github.io/docs/manual/timout.html>
+# which are treated as keywords in csound.py.
 OPCODES = set((
     'ATSadd',
     'ATSaddnz',
@@ -135,6 +116,7 @@
     'OSCinit',
     'OSClisten',
     'OSCsend',
+    'S',
     'a',
     'abs',
     'active',
@@ -151,7 +133,7 @@
     'ampmidid',
     'areson',
     'aresonk',
-    'array',
+    'array', # Deprecated <http://csound.github.io/docs/manual/array.html>
     'atone',
     'atonek',
     'atonex',
@@ -163,9 +145,9 @@
     'bbcuts',
     'betarand',
     'bexprnd',
-    'bformdec',
+    'bformdec', # Deprecated <https://csound.github.io/docs/manual/bformdec.html>
     'bformdec1',
-    'bformenc',
+    'bformenc', # Deprecated <https://csound.github.io/docs/manual/bformenc.html>
     'bformenc1',
     'binit',
     'biquad',
@@ -192,9 +174,11 @@
     'cent',
     'centroid',
     'ceps',
+    'cepsinv',
    #'cggoto',
     'chanctrl',
     'changed',
+    'changed2',
     'chani',
     'chano',
     'chebyshevpoly',
@@ -219,18 +203,18 @@
     'clockon',
     'cmplxprod',
    #'cngoto',
+   #'cnkgoto',
     'comb',
     'combinv',
     'compilecsd',
     'compileorc',
     'compilestr',
     'compress',
+    'compress2',
     'connect',
     'control',
     'convle',
     'convolve',
-    'copy2ftab',
-    'copy2ttab',
     'copya2ftab',
     'copyf2array',
     'cos',
@@ -286,6 +270,7 @@
     'deltapxw',
     'denorm',
     'diff',
+    'directory',
     'diskgrain',
     'diskin',
     'diskin2',
@@ -338,6 +323,7 @@
     'filelen',
     'filenchnls',
     'filepeak',
+    'filescal',
     'filesr',
     'filevalid',
     'fillarray',
@@ -385,6 +371,7 @@
     'fprints',
     'frac',
     'fractalnoise',
+    'framebuffer',
     'freeverb',
     'ftchnls',
     'ftconv',
@@ -400,6 +387,7 @@
     'ftmorf',
     'ftresize',
     'ftresizei',
+    'ftsamplebank',
     'ftsave',
     'ftsavek',
     'ftsr',
@@ -417,6 +405,7 @@
     'getcfg',
     'getcol',
     'getrow',
+    'getseed',
     'gogobel',
    #'goto',
     'grain',
@@ -432,7 +421,7 @@
     'hdf5write',
     'hilbert',
     'hrtfearly',
-    'hrtfer',
+    'hrtfer', # Deprecated <https://csound.github.io/docs/manual/hrtfer.html>
     'hrtfmove',
     'hrtfmove2',
     'hrtfreverb',
@@ -442,7 +431,6 @@
     'hvs2',
     'hvs3',
     'i',
-    'iceps',
    #'igoto',
     'ihold',
     'imagecreate',
@@ -483,9 +471,8 @@
     'jspline',
     'k',
    #'kgoto',
-    'ktableseg',
+    'ktableseg', # Deprecated <https://csound.github.io/docs/manual/ktableseg.html>
     'lenarray',
-    'lentab',
     'lfo',
     'limit',
     'line',
@@ -533,9 +520,8 @@
     'lpshold',
     'lpsholdp',
     'lpslot',
-   #'lua_exec',
-    'lua_ikopcall',
-   #'lua_opdef',
+    'lua_exec',
+    'lua_opdef',
     'mac',
     'maca',
     'madsr',
@@ -553,7 +539,6 @@
     'maxaccum',
     'maxalloc',
     'maxarray',
-    'maxtab',
     'mclock',
     'mdelay',
     'median',
@@ -590,13 +575,13 @@
     'minaccum',
     'minarray',
     'mincer',
-    'mintab',
     'mirror',
     'mode',
     'modmatrix',
     'monitor',
     'moog',
     'moogladder',
+    'moogladder2',
     'moogvcf',
     'moogvcf2',
     'moscil',
@@ -604,12 +589,24 @@
     'mp3in',
     'mp3len',
     'mp3nchnls',
+    'mp3scal',
+    'mp3scal_check',
+    'mp3scal_load',
+    'mp3scal_load2',
+    'mp3scal_play',
+    'mp3scal_play2',
     'mp3sr',
     'mpulse',
     'mrtmsg',
     'multitap',
     'mute',
+    'mvchpf',
+    'mvclpf1',
+    'mvclpf2',
+    'mvclpf3',
+    'mvclpf4',
     'mxadsr',
+    'nchnls_hw',
     'nestedap',
     'nlalp',
     'nlfilt',
@@ -632,6 +629,7 @@
     'octmidib',
     'octmidinn',
     'octpch',
+    'olabuffer',
    #'opcode',
     'oscbnk',
     'oscil',
@@ -684,12 +682,14 @@
     'pan',
     'pan2',
     'pareq',
+    'part2txt',
     'partials',
     'partikkel',
     'partikkelget',
     'partikkelset',
     'partikkelsync',
     'passign',
+    'paulstretch',
     'pcauchy',
     'pchbend',
     'pchmidi',
@@ -723,8 +723,8 @@
     'pol2rect',
     'polyaft',
     'polynomial',
-    'pop',
-    'pop_f',
+    'pop', # Deprecated <https://csound.github.io/docs/manual/pop.html>
+    'pop_f', # Deprecated <https://csound.github.io/docs/manual/pop_f.html>
     'port',
     'portk',
     'poscil',
@@ -751,8 +751,8 @@
     'ptableiw',
     'ptablew',
     'ptrack',
-    'push',
-    'push_f',
+    'push', # Deprecated <https://csound.github.io/docs/manual/push.html>
+    'push_f', # Deprecated <https://csound.github.io/docs/manual/push_f.html>
     'puts',
     'pvadd',
     'pvbufread',
@@ -885,12 +885,12 @@
     'pylexec',
     'pylexeci',
     'pylexect',
-   #'pylrun',
-   #'pylruni',
-   #'pylrunt',
-   #'pyrun',
-   #'pyruni',
-   #'pyrunt',
+    'pylrun',
+    'pylruni',
+    'pylrunt',
+    'pyrun',
+    'pyruni',
+    'pyrunt',
     'qinf',
     'qnan',
     'r2c',
@@ -934,7 +934,7 @@
     'rfft',
     'rifft',
    #'rigoto',
-    'rireturn',
+   #'rireturn',
     'rms',
     'rnd',
     'rnd31',
@@ -947,7 +947,6 @@
     'sandpaper',
     'scale',
     'scalearray',
-    'scalet',
     'scanhammer',
     'scans',
     'scantable',
@@ -956,8 +955,8 @@
     'schedkwhennamed',
     'schedule',
     'schedwhen',
-   #'scoreline',
-   #'scoreline_i',
+    'scoreline',
+    'scoreline_i',
     'seed',
     'sekere',
     'semitone',
@@ -1028,8 +1027,8 @@
     'socksend',
     'socksends',
     'soundin',
-    'soundout',
-    'soundouts',
+    'soundout', # Deprecated <https://csound.github.io/docs/manual/soundout.html>
+    'soundouts', # Deprecated <https://csound.github.io/docs/manual/soundouts.html>
     'space',
     'spat3d',
     'spat3di',
@@ -1049,7 +1048,7 @@
     'sprintfk',
     'spsend',
     'sqrt',
-    'stack',
+    'stack', # Deprecated <https://csound.github.io/docs/manual/stack.html>
     'statevar',
     'stix',
     'strcat',
@@ -1086,7 +1085,6 @@
     'subinstrinit',
     'sum',
     'sumarray',
-    'sumtab',
     'svfilter',
     'syncgrain',
     'syncloop',
@@ -1096,7 +1094,7 @@
     'tab',
     'tab2pvs',
     'tab_i',
-    'tabgen',
+    'tabifd',
     'table',
     'table3',
     'table3kt',
@@ -1122,15 +1120,12 @@
     'tablewkt',
     'tablexkt',
     'tablexseg',
-    'tabmap',
-    'tabmap_i',
     'tabmorph',
     'tabmorpha',
     'tabmorphak',
     'tabmorphi',
     'tabplay',
     'tabrec',
-    'tabslice',
     'tabsum',
     'tabw',
     'tabw_i',
@@ -1210,6 +1205,7 @@
     'unirand',
     'unwrap',
     'upsamp',
+    'urandom',
     'urd',
     'vactrol',
     'vadd',
